lib/src/worker/worker_connection.dart

```diff
@@ -24,6 +26,12 @@ abstract class WorkerConnection {
 }
 
 abstract class AsyncWorkerConnection implements WorkerConnection {
+  /// Creates a [StdAsyncWorkerConnection], unless [sendPort] is specified, in
+  /// which case creates a [SendPortAsyncWorkerConnection].
+  factory AsyncWorkerConnection([SendPort sendPort]) => sendPort == null
```

Comment:
nit: prefer named optional args to positional ones (they scale better, without breaking changes)

Reply:
Done, and passed through `inputStream` and `outputStream` for good measure.